neurots/generate/grower.py

```diff
@@ -198,17 +200,20 @@ def grow(self):
 
     def _post_grow(self):
         """Actions after the morphology has been grown and before its diametrization."""
+        # ensures section.id are consistent with morphio loader
+        self.neuron = Morphology(self.neuron)
 
     def _init_diametrizer(self, external_diametrizer=None):
         """Set a diametrizer function."""
-        if self.input_distributions["diameter"]["method"] == "default":
+        if self.input_distributions["diameter"]["method"] == "no_diameters":
             self._diametrize = lambda: None
             L.warning("No diametrizer provided, so neurons will have default diameters.")
         else:
             if self.input_distributions["diameter"]["method"] == "external":
-                if external_diametrizer is None:
-                    raise Exception("Please provide an external diametrizer!")
```

Comment:
What happens now when `external_diametrizer is None`?

Reply:
I noticed it was duplicate of a check in class __init__ just above

Reply:
Ah yes indeed!